src/engine/SCons/Tool/MSCommon/vc.py

```diff
@@ -344,29 +347,99 @@ def find_batch_file(env,msvc_version,host_arch,target_arch):
 
 __INSTALLED_VCS_RUN = None
 
-def cached_get_installed_vcs():
+def _check_cl_exists_in_vc_dir(env, vc_dir, msvc_version):
+    
+    # determine if there is a specific target platform we want to build for and
+    # use that to find a list of valid VCs, default is host platform == target platform
+    # and same for if no env is specified to extract target platform from
+    if env:
+        (host_platform, target_platform, req_target_platform) = get_host_target(env)
+    else:
+        host_platform = platform.machine().lower()
+        target_platform = host_platform
+
+    ver_num = float(get_msvc_version_numeric(msvc_version))
+
+    # make sure the cl.exe exists meaning the tool is installed
+    if ver_num > 14:
+        # 2017 and newer allowed multiple versions of the VC toolset to be installed at the same time.
+        # Just get the default tool version for now
+        #TODO: support setting a specific minor VC version
+        try:
+            f = open(os.path.join(vc_dir, r'Auxiliary\Build\Microsoft.VCToolsVersion.default.txt'))
+            vc_specific_version = f.readlines()[0].strip()
+        except OSError: 
+            return False
+        except IndexError:
+            return False
+
+        if host_platform in ('amd64','x86_64'):
+            host_dir = "Hostx64"
+        elif host_platform in ('i386','i686','x86'):
+            host_dir = "Hostx86"
+        else:
+            return False
+
+        if target_platform in ('amd64','x86_64'):
+            target_dir = "x64"
+        elif target_platform in ('i386','i686','x86'):
+            target_dir = "x86"
+        else:
+            return False
```

Comment:
should we also have ARM here?

Reply:
yes, still need to do that, I don't have access to a machine that has visual studio installed to help me do it atm

Reply:
yeah they are version specific, so it would require two dicts, but its still makes the code more organized